src/box/sql/build.c

```diff
@@ -1125,34 +1136,72 @@ vdbe_emit_ck_constraint_create(struct Parse *parser,
 	struct Vdbe *v = sqlGetVdbe(parser);
 	assert(v != NULL);
 	/*
-	 * Occupy registers for 5 fields: each member in
-	 * _ck_constraint space plus one for final msgpack tuple.
+	 * Occupy registers for 20 fields: each member in _func space plus one
+	 * for final msgpack tuple.
 	 */
-	int ck_constraint_reg = sqlGetTempRange(parser, 7);
-	sqlVdbeAddOp2(v, OP_SCopy, reg_space_id, ck_constraint_reg);
-	sqlVdbeAddOp4(v, OP_String8, 0, ck_constraint_reg + 1, 0,
-		      sqlDbStrDup(db, ck_def->name), P4_DYNAMIC);
-	sqlVdbeAddOp2(v, OP_Bool, false, ck_constraint_reg + 2);
-	sqlVdbeAddOp4(v, OP_String8, 0, ck_constraint_reg + 3, 0,
-		      ck_constraint_language_strs[ck_def->language], P4_STATIC);
-	sqlVdbeAddOp4(v, OP_String8, 0, ck_constraint_reg + 4, 0,
-		      sqlDbStrDup(db, ck_def->expr_str), P4_DYNAMIC);
-	sqlVdbeAddOp2(v, OP_Bool, true, ck_constraint_reg + 5);
-	sqlVdbeAddOp3(v, OP_MakeRecord, ck_constraint_reg, 6,
-		      ck_constraint_reg + 6);
-	const char *error_msg =
-		tt_sprintf(tnt_errcode_desc(ER_CONSTRAINT_EXISTS),
-			   constraint_type_strs[CONSTRAINT_TYPE_CK],
-			   ck_def->name, space_name);
-	if (vdbe_emit_halt_with_presence_test(parser, BOX_CK_CONSTRAINT_ID, 0,
-					      ck_constraint_reg, 2,
-					      ER_CONSTRAINT_EXISTS, error_msg,
-					      false, OP_NoConflict) != 0)
+	int regs = sqlGetTempRange(parser, 20);
+	char *func_name = sqlMPrintf(db, "check_%s_%s", space_name,
+				     ck_def->name);
+	int name_reg = regs + BOX_FUNC_FIELD_NAME;
+	sqlVdbeAddOp4(v, OP_String8, 0, name_reg, 0, func_name, P4_DYNAMIC);
+	const char *err = tt_sprintf("Function for the check constraint '%s' "
+		"with name '%s' already exists", ck_def->name, func_name);
+	if (vdbe_emit_halt_with_presence_test(parser, BOX_FUNC_ID, 2, name_reg,
```

Comment:
Might be better for readibility to do `2` -> `BOX_FUNC_FIELD_NAME` and `err, 0` -> `err, false` since the next to last arg is a boolean.

Reply:
Thanks, fixed!